src/main/java/org/genomicsdb/importer/MultiChromosomeIterator.java

```diff
@@ -64,8 +64,12 @@
         final SAMSequenceDictionary contigDictionary = header.getSequenceDictionary();
         for (ChromosomeInterval currInterval : chromosomeIntervals)
             if (contigDictionary.getSequenceIndex(currInterval.getContig()) != -1) this.chromosomeIntervals.add(currInterval);
-        if (this.chromosomeIntervals.size() > 0) {
-            ChromosomeInterval currInterval = this.chromosomeIntervals.get(0);
+        //move to next chromosome and iterate
+        //It's possible that the reader has no record for the first contig, but could have records
+        //for subsequent contigs
+        idxInIntervalList = 0;
+        while (idxInIntervalList < this.chromosomeIntervals.size() && !hasNext()) {
+            ChromosomeInterval currInterval = this.chromosomeIntervals.get(idxInIntervalList++);
```

Comment:
Looks like this loop will create an iterator after the first contig with no records. Is it possible for multiple consecutive contigs to not have records?